include/deal.II/lac/trilinos_tpetra_vector.templates.h

```diff
@@ -643,8 +750,45 @@ namespace LinearAlgebra
 
     template <typename Number>
     void
-    Vector<Number>::compress(const VectorOperation::values /*operation*/)
-    {}
+    Vector<Number>::compress(const VectorOperation::values operation,
+                             const IndexSet               &locally_owned)
+    {
+      if (!compressed)
+        {
+          Tpetra::CombineMode tpetra_operation = Tpetra::ZERO;
+          if (operation == VectorOperation::insert)
+            tpetra_operation = Tpetra::INSERT;
+          else if (operation == VectorOperation::add)
+            tpetra_operation = Tpetra::ADD;
+          else
+            Assert(false, ExcNotImplemented());
+
+          if (locally_owned != IndexSet())
+            {
+              locally_owned_map.reset();
+              locally_owned_map =
+                locally_owned.make_tpetra_map_rcp(mpi_comm(), false);
```

Comment:
For this whole block, I don't quite understand why it is necessary. Why would one provide a `locally_owned` index set that is different from the index set of locally owned elements already stored by the vector? Why can't we just always use that? Every vector is created with a set of locally owned elements, after all, or do I misinterpret? What situations can we encounter where a vector does not know its locally owned elements and need to be told about them?

Reply:
I just wanted to cover the case where the vector was created without a locally owned index set.
If you say that vectors are always created with a locally owned index set, then I remove this.

Reply:
So, I just remove this

Reply:
A vector needs to know which element it stores locally -- that's the locally owned index set. So yes, a vector always has to have that. (If the vector operates on only one process, then the set of locally owned elements is simply one that contains *all* of the vector.)